pulsar/consumer_impl.go

```diff
@@ -423,6 +423,11 @@ func (c *consumer) Unsubscribe() error {
 }
 
 func (c *consumer) Receive(ctx context.Context) (message Message, err error) {
+	start := time.Now()
+	defer func() {
+		now := time.Now().UnixNano()
+		c.metrics.RecvTime.Observe(float64(now-start.UnixNano()) / 1.0e9)
```

Comment:
Golang has a more standard way to calculate since time in seconds or any time unit. Here is an example.

time.Since(startTime).Seconds()